### PR TITLE
:sparkles: Manage Security Groups using OpenStackClusterTemplate

### LINKED ISSUE #74: Manage Security Groups using OpenStackClusterTemplate
/kind feature

**Describe the solution you'd like**
With CAPO v0.10.0 and corresponding apiVersion v1beta1 it should be possible to manage security groups on the fly in an Openstack project. Since we have some required security groups e.g. for Cilium, we have to prepare them beforehand. This can and should now be done using CAPI.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Kubernetes version: (use `kubectl version`)
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by michal-gubricky**
**What this PR does / why we need it**:
This PR adds security rules for the Cilium into the default security groups for control-plane and worker nodes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #74 

**Special notes for your reviewer**:
This is how security groups for control-plane and worker nodes look like now:
sec-group-contorlplane
sec-groups-worker

### CODE REVIEW

**Review by DEiselt — COMMENTED** [review]
Nice work, i am unfortunately unable to test this due to issues with the infrastructure providers i have access to.

**Review by DEiselt — APPROVED** [review]
Was able to test now.

File: providers/openstack/scs/cluster-class/templates/openstack-cluster-template.yaml
Comment (chess-knight) [inline providers/openstack/scs/cluster-class/templates/openstack-cluster-template.yaml:1]: You also want to remove the patch for the `allowAllInClusterTraffic` when the `openstack_security_groups` are used.

**Review by chess-knight — CHANGES_REQUESTED** [review]

File: providers/openstack/scs/cluster-class/templates/openstack-cluster-template.yaml
Comment (michal-gubricky) [reply providers/openstack/scs/cluster-class/templates/openstack-cluster-template.yaml:1]: Thanks, I will take a look at it

**Review by chess-knight — APPROVED** [review]
LGTM

**Pull request merged**